src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -1178,30 +1220,30 @@ object LazyList extends SeqFactory[LazyList] {
   def from[A](coll: collection.IterableOnce[A]): LazyList[A] = coll match {
     case lazyList: LazyList[A]    => lazyList
     case _ if coll.knownSize == 0 => empty[A]
-    case _                        => newLL(stateFromIterator(coll.iterator))
+    case _                        => newLL(eagerFromIterator(coll.iterator))
   }
 
-  def empty[A]: LazyList[A] = _empty
+  def empty[A]: LazyList[A] = Empty
 
-  /** Creates a State from an Iterator, with another State appended after the Iterator
-    * is empty.
-    */
-  private def stateFromIteratorConcatSuffix[A](it: Iterator[A])(suffix: => State[A]): State[A] =
-    if (it.hasNext) sCons(it.next(), newLL(stateFromIteratorConcatSuffix(it)(suffix)))
+  /** Creates a LazyList with the elements of an iterator followed by a LazyList suffix.
+   *  Eagerly evaluates the first element.
+   */
+  private def eagerPrependIterator[A](it: Iterator[A])(suffix: => LazyList[A]): LazyList[A] =
+    if (it.hasNext) eagerCons(it.next(), newLL(eagerPrependIterator(it)(suffix)))
     else suffix
 
-  /** Creates a State from an IterableOnce. */
-  private def stateFromIterator[A](it: Iterator[A]): State[A] =
-    if (it.hasNext) sCons(it.next(), newLL(stateFromIterator(it)))
-    else State.Empty
+  /** Creates a LazyList from an Iterator. Eagerly evaluates the first element. */
+  private def eagerFromIterator[A](it: Iterator[A]): LazyList[A] =
+    if (it.hasNext) eagerCons(it.next(), newLL(eagerFromIterator(it)))
+    else Empty
 
   override def concat[A](xss: collection.Iterable[A]*): LazyList[A] =
     if (xss.knownSize == 0) empty
-    else newLL(concatIterator(xss.iterator))
+    else newLL(eagerConcatIterators(xss.iterator))
 
-  private def concatIterator[A](it: Iterator[collection.Iterable[A]]): State[A] =
-    if (!it.hasNext) State.Empty
-    else stateFromIteratorConcatSuffix(it.next().iterator)(concatIterator(it))
+  private def eagerConcatIterators[A](it: Iterator[collection.Iterable[A]]): LazyList[A] =
```

Comment:
same naming concern (sorry for saying this so much, I just don't want to miss any)